### PR TITLE
Added Testing CI

### PR DESCRIPTION
**Pull request by FredAdiv**
@szabgab

### CODE REVIEW

File: .github/workflows/test_ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This workflow will install Python dependencies and run tests with a single version of Python for Penn repository
+
+name: Testing Penn
+
+on:
+  push:
+    branches: [ "master" ]
```
Comment (szabgab) [inline .github/workflows/test_ci.yml:7]: I'd remove the rows restricting the CI to the master branch so other branches would be tested as well.

File: .github/workflows/test_ci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  pull_request:
+    branches: [ "master" ]
```
Comment (szabgab) [inline .github/workflows/test_ci.yml:9]: I'd remove this too.

File: .github/workflows/test_ci.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        pip install -r requirements.txt && pip install -e .
+        pip install pytest
+    - name: Test with pytest
+      run: |
+        pytest
```
Comment (szabgab) [inline .github/workflows/test_ci.yml:31]: I'd run `pytest -svv` to make the output more verbose.

File: .github/workflows/test_ci.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  pull_request:
+    branches: [ "master" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (szabgab) [inline .github/workflows/test_ci.yml:17]: I'd probably add a matrix here to prepare running the tests on multiple versions of python and I'd check if the tests pass on macOS and Windows as well.

**Comment by szabgab** [comment]
@FredAdiv  You sent this PR from your "master" branch. It would be have been better to create a separate branch, make the changes there and send the PR from there.

**Comment by FredAdiv** [comment]
Opened the PR from a new CI branch. Closing this one

**Pull request closed without merging**